src/main/java/seedu/address/logic/parser/MailXCommandParser.java

```diff
@@ -24,26 +25,41 @@
     public MailXCommand parse(String args) throws ParseException {
         requireNonNull(args);
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_EMAIL, PREFIX_MOD, PREFIX_GROUP);
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_EMAIL,
+                        PREFIX_MOD, PREFIX_GROUP, PREFIX_STUDENT_NUMBER);
 
-        MailMDescriptor mailMDescriptor = new MailMDescriptor();
+        System.out.println(argMultimap.getPreamble());
+
+        if ((argMultimap.getPreamble().trim().isEmpty() && argMultimap.isEmpty())
+                || !argMultimap.getPreamble().trim().isEmpty()
+                || argMultimap.isEmpty()
+                || args.trim().isEmpty()) {
```

Comment:
I think there should be a way to split this up into individual booleans so they are easier to check and more readable, what do you think?
Something like, hasPreamble, isMultimapEmpty, etc